java/connector-framework-internal/src/main/java/org/identityconnectors/framework/impl/api/local/ObjectPool.java

```diff
@@ -209,21 +229,39 @@ public boolean isShutdown() {
         return isShutdown;
     }
 
-    /**
+    public boolean isDisposing() {
+		return isDisposing;
+	}
+
+	public String getPoolName() {
+		return poolName;
+	}
+
+	public void setPoolName(String poolName) {
+		this.poolName = poolName;
+	}
+
+	/**
      * Return an object to the pool
      *
      * @param pooled
      */
     private void returnObject(PooledObject pooled) throws InterruptedException {
-        if (isShutdown() || poolConfiguration.getMaxIdle() < 1) {
+        if (isShutdown() || isDisposing() || poolConfiguration.getMaxIdle() < 1) {
             dispose(pooled);
+        	if (LOG.isOk()) {

```

Comment:
This if is redundant, should be avoided

Reply:
Right. Remnants of (many) previous experiments.